### ISSUE TITLE
Can this plugin works with react-adobe-animate

### ISSUE THREAD

**Issue by engrsandeep** [CurriculumAssociates/createjs-accessibility#40]
Hi,

We uses **react-adobe-animat**e plugin. This plugn does not expose stage but access to root container. 
Can you please help.

@mmoss @slahav @acejam @Pratikh @mattbargar-cainc 

Regards,
Sandeep

**Comment by mnewcomb314** [comment]
Hello!
I took a look at react-adobe-animate this morning.  While marking the stage as a private member (https://github.com/bibixx/react-adobe-animate/blob/master/src/AnimateCC/AnimateCC.tsx#L19) means that library doesn't intend to allow outside access to the stage.  In order to prevent accidental writes to that field, that makes sense.  But CAM needs the value for the stage in order to get setup.  So, I would recommend opening a PR with that project to add a public getter for the stage.  That will allow your project to get a reference to the AnimateCC component.  Then once the component is mounted you can call `AccessibilityModule.setupStage` and `AccessibilityModule.register` with the stage and relevant DisplayObjects.

**Comment by engrsandeep** [comment]
Thanks @mnewcomb314  for quick reply.

**Comment by engrsandeep** [comment]
@mnewcomb314 Can you please help. i am getting following errors
<img width="891" alt="Screenshot 2021-02-12 at 10 30 47 AM" src="https://user-images.githubusercontent.com/10007250/107732082-76d3c000-6d1d-11eb-9504-b3af91b0f07a.png">

I have only aded this code for now...
    const onReady = () => {
      // discussed in next step
      AccessibilityModule.register({
        displayObject: exportRoot,
        role: AccessibilityModule.ROLES.NONE,
      });
      stage.accessibilityTranslator.root = exportRoot;
    };
    AccessibilityModule.setupStage(stage, 'cam-test', onReady);

**Comment by mnewcomb314** [comment]
What is your import line?  After updating react-adobe-animate example's package.json to point to that CAM version and installing it for that project, I was able to import and call that function.  The import line I'm using is a copy-and-paste from CAM's test app (https://github.com/CurriculumAssociates/createjs-accessibility-tester/blob/main/src/index.js#L2).  If the import line isn't the source of the issue, if you could setup a minimal test case project for this issue to share, it would be helpful for debugging.  But I also recommend using CAM's test app for reference (https://github.com/CurriculumAssociates/createjs-accessibility-tester).

**Comment by engrsandeep** [comment]
@mnewcomb314 
i have removed  react-adobe-animate plugin, so now loading createjs using import loader. but i am still getting same error. 
I am also using export file created from AnimateCC.  i am setting up sample. but i am getting this error now

<img width="1091" alt="Screenshot 2021-02-13 at 1 25 00 PM" src="https://user-images.githubusercontent.com/10007250/107845081-fa191280-6dfe-11eb-8640-962461b957d1.png">

**Comment by mnewcomb314** [comment]
There looks to be a couple of issues.  From googling the jsx part of the error message, your babel config might be setup to transpile node_modules dependencies.  https://github.com/CurriculumAssociates/createjs-accessibility-tester contains a simple but useful reference for working babel and webpack configs.  The second is that it is looking at src/index.js inside of node_modules/@curriculumassociates/createjs-accessibility/.  The package.json of the 1.0.0-prerelease.9 release has its "main" field pointed at `dist/createjs-accessibility.js`, which means that dist file instead of src/index.js should be used when building consuming projects.  If the package.json of this module in your node_modules directory doesn't point to that dist file, then please run `rm -rf node_modules/ && npm cache clean --force` followed by `npm install`.  If you need further assistance, a sample minimal test case project displaying this issue that we can both access would be helpful to debug the issues since I'm not able to reproduce these issues with either the test app or modifications to react-adobe-animate's example app.

**Comment by engrsandeep** [comment]
Thanks @mnewcomb314. issue is fixed.
One query. can we update accessibility text dynamically for an element?

**Comment by mnewcomb314** [comment]
Yes!  There are two things that need to happen for this.  The first is a little setup of having this module do its render pass of the DOM when the stage is redrawn, such as https://github.com/CurriculumAssociates/createjs-accessibility-tester/blob/main/src/index.js#L17-L20.  The second part is that most properties for a role have getters and setters that will be used to populate the various HTML attributes.  `text` is a little bit of an odd-ball in that it is just a public field on the AccessibilityObject (https://github.com/CurriculumAssociates/createjs-accessibility/blob/main/src/RoleObjects/AccessibilityObject.js#L40-L44) rather than having a getter/setter.  So you can do something along the lines of `displayObjectInstace.accessible.text = "new value";` in your code then on the next render pass, the new value will be in the DOM.

**Comment by engrsandeep** [comment]
Thank you @mnewcomb314. i will give it a try today.

**Comment by engrsandeep** [comment]
@mnewcomb314  This is working perfect. Thanks. 

I have one other query. Role button specifically should get focus even screen reader not opened i think as per WCAG guidelines. is that achieveable.

**Comment by mnewcomb314** [comment]
All enabled and interactive elements included in the tab order or with tabIndex set to -1 can receive focus programmatically.  The AccessibilityObject base class provides a requestFocus function (https://github.com/CurriculumAssociates/createjs-accessibility/blob/main/src/RoleObjects/AccessibilityObject.js#L126-L130), so you can do something along the lines of `displayObject.accessible.requestFocus();`.  That function also deals with properties being updated that should allow for an element to be focusable, but the render pass after those properties were updated not having occurred yet, though it still assumes that a render pass to put the element into the document's DOM has occurred.

**Comment by drosenberg123** [comment]
> @mnewcomb314 This is working perfect. Thanks.
> 
Closing as resolved.
